docs/v1/p4runtime-id-notes.md

```diff
@@ -0,0 +1,487 @@
+# P4Runtime numeric id notes
+
+There are many kinds of numeric identifiers used in the P4Runtime API
+specification.  While they are all described in the specification and
+accompanying Protobuf `.proto` files, this article attempts to collect
+in one place notes on those identifiers, including:
+
++ Which software entity (or person) chooses their values
++ When the values are chosen
++ What is their scope in "space" and time in which they must be unique
+
+This document was written while referring to the P4Runtime API
+specification as of this commit of the Github repository
+https://github.com/p4lang/p4runtime
+
+```
+commit c9cd4af17c2d562d6e4ec760237f16fea9558fcb
+Author: Chris Sommers <31145757+chrispsommers@users.noreply.github.com>
+Date:   Fri Mar 27 13:43:31 2020 -0700
+```
+
+This is after version 1.1.0 of the specification was released, and it
+is expected that version 1.2.0 will be released soon after this time,
+with very few (if any) changes from the version above.
+
+Here is a complete list of Google Protobuf `.proto` files in this
+version of the p4lang/p4runtime repository:
+
+```bash
+$ find . -name '*.proto'
+./proto/p4/v1/p4runtime.proto
+./proto/p4/v1/p4data.proto
+./proto/p4/config/v1/p4info.proto
+./proto/p4/config/v1/p4types.proto
+```
+
+And here are the number of occurrences of `id`, either as a word by
+itself, or with `_` before it, in each of the files above.  These
+restrictions are made to avoid matching against words like `valid` or
+`bidirectional` in the comments.  As of the referenced version of the
+files, it appears that all occurrences of numeric id values are either
+a word by themselves, or have an underscore character just before
+`id`.
+
+```bash
+$ find . -name '*.proto' | xargs egrep -c '(\bid\b|_id\b)'
+./proto/p4/v1/p4runtime.proto:56
+./proto/p4/v1/p4data.proto:0
+./proto/p4/config/v1/p4info.proto:22
+./proto/p4/config/v1/p4types.proto:0
+```
+
+
+# Numeric ids in the P4Info message
+
+These are all defined in the file `proto/p4/config/v1/p4info.proto`.
+
+
+## Numeric ids that are unique within the P4Info top level scope
+
+The messages listed below are all of those that are sub-messages of
+the `P4Info` message, and they contain a `Preamble` sub-message.  Each
+`Preamble` message contains an `id` field, as well as several others
+like `name`, `alias`, and `annotations`.  See the `Preamble` message
+definition for the full list.
+
+Section 6.3 "ID Allocation for P4Info Objects" of the P4Runtime API
```

Comment:
suggestion: you can put a link to the section here, this way you ensure you won't need to update the section number in the future.

Reply:
Do you mean a link to the section heading of the HTML version of the latest auto-generated version, e.g. this?  https://p4.org/p4runtime/spec/master/P4Runtime-Spec.html#sec-id-allocation

Reply:
Yes, that's what I meant